nautobot/dcim/models/devices.py

```diff
@@ -998,3 +1025,74 @@ def to_csv(self):
             self.domain,
             self.master.name if self.master else None,
         )
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "dynamic_groups",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "statuses",
+    "webhooks",
+)
+class DeviceRedundancyGroup(PrimaryModel, StatusModel):
+    """
+    A DeviceRedundancyGroup represents a logical grouping of physical hardware for the purposes of high-availability.
+    """
+
+    name = models.CharField(max_length=100, unique=True)
+    slug = AutoSlugField(populate_from="name")
+    description = models.CharField(max_length=200, blank=True)
+
+    failover_strategy = models.CharField(
+        max_length=50,
+        blank=True,
+        choices=DeviceRedundancyGroupFailoverStrategyChoices,
+        verbose_name="Failover strategy",
+    )
+
+    comments = models.TextField(blank=True)
+
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        default=None,
+        blank=True,
+        null=True,
+    )
+
+    clone_fields = [
+        "failover_strategy",
+        "status",
+        "secrets_group",
+    ]
+
+    csv_headers = ["name", "failover_strategy", "status", "secrets_group", "comments"]
+
+    class Meta:
+        ordering = ("name",)
+
+    @property
+    def members_sorted(self):
+        return self.members.order_by("device_redundancy_group_priority")
+
+    def __str__(self):
+        return self.name
+
+    def natural_key(self):
+        return (self.name,)
```

Comment:
It looks like you forgot to include adding `get_by_natural_key()` method to the queryset/manager for this model, which means you'll need to subclass `RestrictedQuerySet` and then overload `objects =` on this model. 😶

Reply:
Addressed this @jathanism. Please review for any other show stoppers.